packages/alchemy/src/react/hooks/useSignMessage.ts

```diff
@@ -0,0 +1,72 @@
+"use client";
+
+import { useMutation, type UseMutateFunction } from "@tanstack/react-query";
+import type { Hex, SignableMessage } from "viem";
+import { useAlchemyAccountContext } from "../context.js";
+import { useBundlerClient } from "./useBundlerClient.js";
+import { type UseSmartAccountClientResult } from "./useSmartAccountClient.js";
+
+export type SignedMessageInfo = {
+  signature: Hex;
+  isValid: boolean;
+};
+
+export type SignMessageArgs = { message: SignableMessage };
+
+export type UseSignMessageArgs = {
+  client?: UseSmartAccountClientResult["client"];
+};
+
+export type UseSignMessageResult = {
+  signMessage: UseMutateFunction<
+    SignedMessageInfo,
+    Error,
+    SignMessageArgs,
+    unknown
+  >;
+  signedMessageInfo: SignedMessageInfo | undefined;
+  isSigningMessage: boolean;
+  error: Error | null;
+};
+
+export function useSignMessage({
+  client,
+}: UseSignMessageArgs): UseSignMessageResult {
+  const { queryClient } = useAlchemyAccountContext();
+  const bundlerClient = useBundlerClient();
+
+  const {
+    mutate: signMessage,
+    data: signedMessageInfo,
```

Comment:
I wonder if it makes sense to have this hook also return `signMessageAsync` so that a user can do `const signature = await signMessageAsync`?

Reply:
agree with this